### PR TITLE
3.6.0 rc2

### PR DESCRIPTION
**Pull request by zjhuntin**

### CODE REVIEW

**Comment by zjhuntin** [comment]
[test]

**Comment by zjhuntin** [comment]
[test]

**Comment by zjhuntin** [comment]
[test katello]

**Comment by zjhuntin** [comment]
[test katello]

**Comment by zjhuntin** [comment]
[test katello]

File: app/services/katello/registration_manager.rb [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      def unregister_host(host, options = {})
+        organization_destroy = options.fetch(:organization_destroy, false)
+        unregistering = options.fetch(:unregistering, false)
+
+        # if the first operation fails, just raise the error since there's nothing to clean up yet.
+        candlepin_consumer_destroy(host.subscription_facet.uuid) if !organization_destroy && host.subscription_facet.try(:uuid)
+
+        # if this fails, there is not much to do about it right now. We can't really re-create the candlepin consumer.
+        # This can be cleaned up later via clean_backend_objects.
+        pulp_consumer_destory(host.content_facet.uuid) if host.content_facet.try(:uuid)
```
Comment (rplevka) [inline app/services/katello/registration_manager.rb]: oooops ;) destory vs destroy

**Pull request merged**